### ISSUE TITLE
Dragon Age™: The Veilguard (1845910)

### ISSUE THREAD

**Issue by Kenyon-J** [ValveSoftware/Proton#8251]
# Compatibility Report
- Name of the game with compatibility issues: DragonAge: Veilguard
- Steam AppID of the game: 1845910

## System Information
- GPU: AMD RX 6700 XT
- Video driver version: fedora-amdgpu
- Kernel version: Linux 6.11.7-300.fc41.x86_64
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/Kenyon-J/cb64ba443888862f2863df6f2cb843eb
- Proton version: Experimental & 9.0-3

## I confirm:
- [ X] that I haven't found an existing compatibility report for this game.
- [ X] that I have checked whether there are updates for my system available.

[steam-1845910.log](https://github.com/user-attachments/files/17787534/steam-1845910.log)

## Symptoms 

"strand hair" toggle on causing game crashes. This may either be a proton or AMD issue - I am not sure and don't have the facilities to test on an intel/nvidia device. I've found numerous reports on internet forums of people discussing this issue but have not seen a ticket raised regarding this just yet- here is one such thread. https://www.reddit.com/r/linux_gaming/comments/1ghmp3x/dragon_age_veilguard_freezing_and_crashing/

some users also suggest it may be Raytracing implementation as well.

## Reproduction

Launch game and play with High settings + strand hair enabled. Crashing typically happens between 30m-2Hr gameplay marks.

**Comment by cpuccino** [comment]
Lighting set to anything but low (the lowest you can go), adds lots of flashing light artifacts everywhere - neve's forehead, harding's face etc

Also stutters really bad after a couple minutes to an hour of playthrough 

4090k, driver 560.35.03, 13900kf - ubuntu 24.10 - wayland - im running proton experimental (for frame gen - even with that, it stutters quite bad (shaders probably as my cpu spikes))

**Comment by etonbears** [comment]
Play-stutter-crash sounds like my experience (Ubuntu 24.04, Proton 9/Experimental, Ryzen 7800X3D, Radeon 7900XT, RADV ).
I used to get  D3D Device-Lost errors as well, but these have gone away, possibly due to game patches.

Eventually, I realised that the issue for me is system memory being spewed into the void. The game plays fine, but chews through system memory until it has about 55GB of 64GB, then stutters badly while trying to get more memory from whatever source it can, eventually leading to Wayland crashing, leaving me at a login screen.

Do you have this memory behaviour as well?

**Comment by cpuccino** [comment]
![image](https://github.com/user-attachments/assets/8f7e1ead-33dc-4de1-995b-001dba6bfc00)

The game crashes randomly with this error, not sure if this is specific to Nvidia or proton Nvidia Driver: `560.35.03`

**Comment by etonbears** [comment]
How about system memory? If you run "top" in a shell/terminal, the game should be the first entry. Does the memory use stabilize as you play, or does it continue to suck up all available memory?

You can also check the VRAM used while playing the game by opening settings from the game menu and looking at the graphical settings tab.

In my case, the game reports using a maximum of ~11GB VRAM, but the system memory use continually increases, which looks like it may be a memory leak.

**Comment by gort818** [comment]
HDR doesn't seem to be working, you can enable it in game, open the HDR calibration but changing the values doesn't have any effect.

Toggling HDR on and off has no effect.
Only HDR game I have that doesn't seem to work with HDR on Linux.

Using latest Gamescope master, Arch Linux 7900XTX.

**Comment by gort818** [comment]
Setting `SteamDeck=1` along with gamescope allows HDR to work.

**Comment by matte-schwartz** [comment]
Yes, to summarize our findings from looking at this over in gamescope:

`SteamDeck=0` will present a broken HDR calibration page (see below) but give you fully working keyboard + mouse input along with a visible cursor

[steam-1845910-steamdeck0.log](https://github.com/user-attachments/files/17988688/steam-1845910-steamdeck0.log)


![Screenshot_20241202_153913-min](https://github.com/user-attachments/assets/26bdd5e6-20e0-47e9-a0d3-2ece1de2e11b)


`SteamDeck=1` will give you a working HDR calibration page (see blow), but it will lock the game to controller input (keyboard still works, but only controller keybinds show) and no cursor is visible

[steam-1845910-steamdeck1.log](https://github.com/user-attachments/files/17988698/steam-1845910-steamdeck1.log)

![dragon-age](https://github.com/user-attachments/assets/f50f56c3-db75-402d-8bc3-de28d968c201)

**Comment by amardhruva** [comment]
Veilguard crashes randomly with DXGI_DEVICE_REMOVED
![image](https://github.com/user-attachments/assets/e451bae6-2591-4a57-96fa-e2bd7d99a556)

OS: arch linux
GPU: AMD 7900 XTX

I also run inside of gamescope with HDR enabled. Command: `MANGOHUD=1 DXVK_HDR=1 gamemoderun gamescope -f --mangoapp -W 3440 -H 1440 -r 165  --hdr-enabled -- %command%`
Relavent kernel logs
[kernelcrash.log](https://github.com/user-attachments/files/18001215/kernelcrash.log)

**Comment by amardhruva** [comment]
> SteamDeck=1

I feel like adding this is the only way to enable hdr as disabling it seems to somehow disable hdr in my machine even though it is checked in the settings. but my keyboard and mouse works though...

**Comment by krhnoo** [comment]
I can get this crash consistently on certain action sequence with this error

<img width="390" height="216" alt="Image" src="https://github.com/user-attachments/assets/eb1c95b3-d72f-4c25-8b3b-b26b54c08f6e" />

Debian sid, Linux 6.15.9-1, Mesa 25.0.7.2, SDR, proton hotfix? suggested by Steam. I'll try lowering settings (disable RT, fancy hair etc) and see if I can get forward on this.

**Comment by Blisto91** [comment]
@krhnoo HI there. Are you able to provide some reproduction steps? What is the action sequence?

It might be interesting to try the game on something else than Proton Hotfix such as 10.0-2 or Experimental. 
If you could attach a Proton log then that would also be great. Running the game with `PROTON_LOG=1 %command%` in the games Steam launch options should make it appear in your users home folder.

**Comment by krhnoo** [comment]
Yeah, there's certain level where doing things in certain order has made it crash three times (pick up a note item and select party member function with tab menu)

I did try proton experimental, but it only postponed the crash slightly, but I'll do a run with logging tomorrow.

E: Actually ran the game and had it crash in under 5 minutes, so here is the log

[steam-1845910.zip](https://github.com/user-attachments/files/21671213/steam-1845910.zip)

**Comment by Blisto91** [comment]
Are you able to share a save game and then steps to reproduce?

**Comment by krhnoo** [comment]
I suppose I could share a save file, not sure if you can load saves from other people on the game being online thing? I'd just rather not post it here as I'm bad at video games 😅  Throw me email krhnoo at proton.me and I'll share it, but the bit is in the crossroads where you have this complicated thing where you have to shoot red lyrium crystals, read a note and evoke companion action by the tab menu, usually the tab action crashed it here.

**Comment by krhnoo** [comment]
Turning details to low and restarting game I was able to get past the crashy bit.

**Comment by masteratt** [comment]
My game also crashes almost instantly (after loading into the gameplay) with Strand Hair enabled, but is completely stable with it disabled. I'm running 4K, everything maxed including RT.

I get DXGI_ERROR_DEVICE_REMOVED pop-up as above, and logs show an NVIDIA Xid 109 — CTX SWITCH TIMEOUT

System:

    Fedora Linux 44

    NVIDIA driver: 610.57.04

    KMD Version: 610.57.04

    CUDA UMD Version: 13.3

    KDE Plasma: 6.7.4

    KDE Frameworks: 6.29.0

    Qt: 6.11.1

    Kernel: 7.1.8-200.fc44.x86_64

    Wayland

    CPU: AMD Ryzen 9 7950X

    RAM: 32 GiB

    GPU: NVIDIA GeForce RTX 4090

    iGPU: Ryzen 9 7950X integrated graphics

    Motherboard: MSI MPG B650 CARBON WIFI (MS-7D74)

    BIOS: 1.L2
